### PR TITLE
Added steps for cluster move operation and added a reference to the new capi in-cluster-ippool solution

### PR DESCRIPTION
**Pull request by arbhoj**
Added steps for cluster move operation and added a reference to the new capi in-cluster-ippool solution.

### CODE REVIEW

File: content/posts/clusterapi-vsphere-nodes-with-defined-ip-range/index.md [hunk 1]
```diff
@@ -247,4 +251,76 @@ export KUBECONFIG=$(pwd)/${CLUSTER_NAME}.conf #Point kubectl to the downloaded k
...
+      end: 10.16.56.92
+    - start: 10.16.56.93
+      end: 10.16.56.93
+  prefix: 20
+  gateway: 10.16.48.1
+  dnsServers: [${DNS_SERVER}]
+EOF
+```
+
+Step 2: Perform activities like upgrade, updates (like node deletion), scaling etc. and update IPPool with any IP that was released. i.e. If any IP address is freed up as a result of this activity, add it as an individual IP to the list of IP addresses in the given pool. They will be used for the next upgrade/update activity that will require additional static IPs for the control plane.
```
Comment (mkamsikad2) [inline content/posts/clusterapi-vsphere-nodes-with-defined-ip-range/index.md:292]: Would recommend:
Step 2: Replace each VSphereMachine by deleting each Machine resource in turn. The capi resources will create a replacement machine which will take an IP address from the IPPool. When all the the machines have been replaced, the IP addresses allocated to the original machines will have been freed up. They can then be added to the IP Pool.

**Comment by msdolbey** [comment]
@arbhoj - I made some grammar changes and just pushed them - if you need any other changes, you'll want to pull.

File: content/posts/clusterapi-vsphere-nodes-with-defined-ip-range/index.md [hunk 2]
```diff
@@ -247,4 +251,76 @@ export KUBECONFIG=$(pwd)/${CLUSTER_NAME}.conf #Point kubectl to the downloaded k
...
+> The **important** point to understand here is that the CAPI controllers only request IPs from the IPPool when provisioning a new VM (i.e. when the VSphereMachine resource is created). Which could be when you are creating, upgrading or scaling the nodes. So, after moving the CAPI resources, start with an IPPool with unused IP addresses and then slowly add them back to the pool as they are released during an upgrade or delete cycle.  
+
+> Note: The following example is assuming that there were 6 IP’s (10.16.56.88-10.16.56.93) allocated/kept aside, for the Control Plane (i.e. 3 for initial deploy and 3 for upgrade, scaling, maintenance etc.). So only the first 3 were assigned to the Control Plane VMs (i.e. 10.16.56.88, 10.16.56.89, 10.16.56.90)
+ 
+
+Step 1: After the cluster is pivoted to the self managed workload cluster, deploy the IPAM manifests as defined in this [section](#steps-to-deploy-capv-with-ipam) and then create the IPPool resource with IPs that are kept aside for the cluster but not allocated to any VM when building the cluster. To keep things simple, create multiple ranges in the IPPool with a single IP in each range (this is not a hard requirement).
+
+```
+e.g.:
```
Comment (msdolbey) [inline content/posts/clusterapi-vsphere-nodes-with-defined-ip-range/index.md:267]: I think the e.g. should either be just taken out or have it above. If someone uses this code block to copy/paste, the e.g. will be there

```suggestion
```

File: content/posts/clusterapi-vsphere-nodes-with-defined-ip-range/index.md [hunk 3]
```diff
@@ -247,4 +251,76 @@ export KUBECONFIG=$(pwd)/${CLUSTER_NAME}.conf #Point kubectl to the downloaded k
...
+  prefix: 20
+  gateway: 10.16.48.1
+  dnsServers: [${DNS_SERVER}]
+EOF
+```
+
+Step 2: Perform activities like upgrade, updates (like node deletion), scaling etc., and update IPPool with any IP that was released, i.e. if any IP address is freed up as a result of this activity, add it as an individual IP to the list of IP addresses in the given pool. They will be used for the next upgrade/update activity that will require additional static IPs for the control plane.
+
+```
+e.g.:
```
Comment (msdolbey) [inline content/posts/clusterapi-vsphere-nodes-with-defined-ip-range/index.md:295]: Same here - I think the e.g. should either be just taken out or have it above. If someone uses this code block to copy/paste, the e.g. will be there

```suggestion
```

**Review by msdolbey — APPROVED** [review]
👍 
I made a couple small nits - other than that, looks good to me:

Mooooooving operations

**Review by arbhoj — COMMENTED** [review]
Looks good

**Review by msdolbey — APPROVED** [review]
👍

File: content/posts/clusterapi-vsphere-nodes-with-defined-ip-range/index.md [hunk 4]
```diff
@@ -247,4 +251,75 @@ export KUBECONFIG=$(pwd)/${CLUSTER_NAME}.conf #Point kubectl to the downloaded k
 kubectl get nodes -o wide #Print the list of nodes along with their IPs
 ```
 
-Well. That's it. Hope you found this useful in setting up a vsphere environment.
+## Steps for Moving to Another Cluster
+ClusterAPI is designed to work in a “Management Cluster > Workload Cluster” topology, which works well with DKP Enterprise. The solution discussed in this blog works well for that classic scenario.
+
+However, if using DKP Essentials with a Self Managed Cluster (or other scenarios where the CAPI resources were moved after the cluster build), the following extra steps need to be performed after moving CAPI resources, as the “IPPool.ipam.metal3.io” resource and other custom resources that keep track of allocated/unallocated IP Addresses are not moved automatically when the CAPI cluster resources are pivoted from the bootstrap cluster to the self managed workload cluster using the `dkp move capi-resources` command:
```
Comment (mmcelroy91) [inline content/posts/clusterapi-vsphere-nodes-with-defined-ip-range/index.md:257]: Self-managed

**Review by mmcelroy91 — APPROVED** [review]

**Review by tillt — APPROVED** [review]

**Review by arbhoj — COMMENTED** [review]
Looks good to me.

**Pull request merged**